polaris-react/src/components/DataTable/DataTable.scss

```
@@ -259,3 +260,15 @@ $breakpoint: 768px;
     background: none;
   }
 }
+
+.FixedFirstColumn {
+  position: absolute;
+
+  background: white;
+
+  z-index: 1;
+  border-collapse: collapse;
+  &.Visible {
```

(This will fix the lint error failing CI)

```suggestion
  
  &.Visible {
```